lasagne/layers/conv.py

```diff
@@ -757,7 +764,18 @@ def get_W_shape(self):
         # first two sizes are swapped compared to a forward convolution
         return (num_input_channels, self.num_filters) + self.filter_size
 
-    def get_output_shape_for(self, input_shape):
+    def get_output_shape_for(self, input_shape, symbolic=False):
+        if self.output_size is not None:
+            if isinstance(self.output_size, T.TensorVariable):
+                if not symbolic:
+                    size = (None, None)
+                else:
+                    size = self.output_size
```

Comment:
I'd prefer to have the `symbolic` distinction in `get_output_for`. It's a nice idea, but you need to read more code to understand it.